### PR TITLE
identity: added context to add key and execute method

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
-  added context to add key and execute method
- removed stored did in identity service

### CODE REVIEW

**Comment by vedhavyas** [comment]
Can we increase the test coverage for this @manuelpolzhofer ?

File: identity/did/identity.go [hunk 1]
```diff
@@ -129,8 +132,28 @@ func logTxHash(tx *types.Transaction) {
...
+	tc, err := contextutil.Account(ctx)
+	if err != nil {
+		return did, err
+	}
+
+	addressByte, err := tc.GetIdentityID()
+	if err != nil {
+		return did, err
+	}
+	did = NewDID(common.BytesToAddress(addressByte))
```
Comment (vedhavyas) [inline identity/did/identity.go]: Do we need to expose `NewDID` ?

File: identity/did/identity.go [hunk 2]
```diff
@@ -129,8 +132,28 @@ func logTxHash(tx *types.Transaction) {
...
+	return did, nil
+
+}
+
+func (i identity) AddKey(ctx context.Context, key Key) (chan *ethereum.WatchTransaction, error) {
```
Comment (vedhavyas) [inline identity/did/identity.go]: add comments pls

File: identity/did/identity.go (hunk 1 above)
Comment (manuelpolzhofer) [reply identity/did/identity.go]: I have to think about how exactly this will replace CentID.

**Comment by manuelpolzhofer** [comment]
@vedhavyas Unit tests are still missing. I will add additional tests?

File: identity/did/identity.go (hunk 2 above)
Comment (manuelpolzhofer) [reply identity/did/identity.go]: I thought we only add comments as required only to the interface method and not to the non-public implementation.

File: identity/did/identity.go (hunk 1 above)
Comment (vedhavyas) [reply identity/did/identity.go]: It should be multi-step
1. lets make CentID type to `common.Address` and rename it to `identity.DID`
2. Change all references of `did.DID` to `identity.DID` in did package and wherever used

File: identity/did/identity.go (hunk 2 above)
Comment (vedhavyas) [reply identity/did/identity.go]: adding comments to interface explains what method should do or return but adding it to the implementation explains implementation specific details.

**Comment by vedhavyas** [comment]
@manuelpolzhofer did you add the tests ?

**Pull request merged**